### ISSUE TITLE
Compile fail when '-mpi' and 'intel' is specified

### ISSUE THREAD

**Issue by lambdalisue** [Amber-MD/cpptraj#270]
The following lines overwrite `CC`, `CXX`, and `FC` automatically

```configure
CC=mpicc
CXX=mpicxx
FC=mpif90
```
https://github.com/Amber-MD/cpptraj/blob/master/configure#L1017

However, if the compiler is `'intel'`, I think the correct one should be

```configure
CC=mpiicc
CXX=mpiicpc
FC=mpiifort
```

So it would be nice if `configure` automatically detect these difference and/or do not overwrite if `CC` was specified by user already like `CC=mpiicc CXX=mpiicpc FC=mpiifort ./configure -mpi -shared intel`

**Comment by hainm** [comment]
hi @lambdalisue,

@drroe  will reply to you soon but I am just curious: what's your case to use `-mpi` with `-shared`. Do you intend to link to your program?

**Comment by lambdalisue** [comment]
@hainm I'm sorry to make you confuse. I wasn't sure about `-mpi` or `-shared` options at that time.
But the point is `-mpi` with `intel`

**Comment by drroe** [comment]
@lambdalisue Good catch, I'll start working on it. Thanks for the report!

**Comment by swails** [comment]
> However, if the compiler is 'intel', I think the correct one should be
> 
> CC=mpiicc
> CXX=mpiicpc
> FC=mpiifort
> So it would be nice if configure automatically detect these difference and/or do not overwrite if CC was specified by user already like CC=mpiicc CXX=mpiicpc FC=mpiifort ./configure -mpi -shared intel

Actually, this is only true for Intel MPI.  If you build mpich2 or mvapich (or something) with the Intel compilers (as I do), the compiler wrappers are still mpicc, mpiCC, and mpif90.  The way AmberTools deals with this is to provide a `-intelmpi` flag to set the compilers to mpiifort, mpiicc, and mpiicpc.  Probably what we should do here.

**Comment by drroe** [comment]
> The way AmberTools deals with this is to provide a -intelmpi flag to set the compilers to mpiifort, mpiicc, and mpiicpc. Probably what we should do here.

Should, and have. If you want to use Intel's MPI, specify '-intelmpi' to configure instead of '-mpi'.

**Comment by drroe** [comment]
Addressed by #274

**Comment by lambdalisue** [comment]
Thanks a lot :-)
